### PR TITLE
Fix compilation errors from mock canvas copy

### PR DESCRIPTION
**Pull request by Piinks**
Follow up PR. The last PR fixed analysis issues, now we are blocked on some compilation issues from copied code we are trying to make public in https://github.com/flutter/flutter/pull/131631

The customer_testing branch and flutter/tests will need to be updated again once this lands.

## Pre-launch Checklist

- [x] I have incremented the package version as appropriate and updated `CHANGELOG.md` with my changes 
- [x] I have added/updated relevant documentation 
- [x] I have run "optimize/organize imports" on all changed files
- [x] I have addressed all analyzer warnings as best I could

### CODE REVIEW

**Comment by Piinks** [comment]
@GroovinChip 😁

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,6 @@
+## [2.0.2]
+* Modified mock_canvas copy from the framework to unblock move to flutter_test
+
 ## [2.0.1]
 * Modified mock_canvas copy from the framework to unblock move to flutter_test
```
Comment (GroovinChip) [inline CHANGELOG.md]: Since both entries are identical, do we need to update the version? If so, let's link to the relevant PR in each entry.

**Review by GroovinChip — CHANGES_REQUESTED** [review]

File: CHANGELOG.md (hunk 1 above)
Comment (Piinks) [reply CHANGELOG.md]: I am fine either way, was just following the instructions:

```html
- [x] I have incremented the package version as appropriate and updated
    `CHANGELOG.md` with my changes <!-- THIS IS REQUIRED -->
```

I'm ok not incrementing it if you are! :)

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,6 @@
+## [2.0.2]
+* Modified mock_canvas copy from the framework to unblock move to flutter_test
+
 ## [2.0.1]
```
Comment (Piinks) [inline CHANGELOG.md]:
```suggestion
## [2.0.1]
```

File: CHANGELOG.md (hunk 1 above)
Comment (GroovinChip) [reply CHANGELOG.md]: Touchè!! I'm fine with leaving it at 2.0.1.

**Review by GroovinChip — APPROVED** [review]
LGTM!

**Comment by GroovinChip** [comment]
@Piinks actually I just noticed that the last PR didn't update the version in the pubspec, so can we actually remove the 2.0.1 changelog entry too?

**Comment by Piinks** [comment]
Sure! One sec!

**Review by GroovinChip — APPROVED** [review]
LGTM

**Comment by GroovinChip** [comment]
@Piinks customer_testing has been updated!

**Pull request merged**